services/notification/notifiers/mixins/message/__init__.py

```diff
@@ -69,6 +72,36 @@ async def create_message(
             f'## [Codecov]({links["pull"]}?dropdown=coverage&src=pr&el=h1) Report',
         ]
 
+        repo = comparison.head.commit.repository
+        owner: Owner = repo.owner
+
+        # Separate PR comment based on plan that can't/won't be tweaked by codecov.yml settings
+        if (
+            owner.plan == BillingPlan.team_monthly.value
+            or owner.plan == BillingPlan.team_yearly.value
+        ):
+            writer_class = TeamPlanWriterSection()
```

Comment:
[nit] this should be in a separate function to make it more explicit we are not using the typical message writing process but a new, independent and dedicated thing to `TeamPlan`